docs/Users/Tutorials/Writing_Bears.rst

```diff
@@ -188,3 +188,78 @@ yield them and write the method as a generator.
     about how bears will look in the future, please read up on
     https://github.com/coala-analyzer/coala/issues/725 or ask us on
     https://gitter.im/coala-analyzer/coala.
+
+Bears Depending on Other Bears
+------------------------------
+
+So we've got a result, but what if we need our Bear to depend on results from
+a different Bear?
+
+Well coala has an efficient dependency management system
+that would run the other Bear before your Bear and get its results for you.
+All you need to do is to tell coala which Bear(s) you want to run before your
+Bear.
+
+So lets see how you could tell coala which Bears to run before yours:
+
+.. code:: python
+
+    from coalib.bears.LocalBear import LocalBear
+    from bears.somePathTo.OtherBear import OtherBear
+
+    class DependentBear(LocalBear):
+
+        def run(self, filename, file, dependency_results):
+            results = dependency_results[OtherBear.__name__]
+
+        @staticmethod
+        def get_dependencies():
+            return [OtherBear]
+
+As we can see we have a `get_dependencies
+<https://coala.readthedocs.org/en/latest/API/coalib.bears.html#coalib.bears.Bear.Bear.get_dependencies>`_
+method which returns a list of bears we wish to depend on.
+In this case it is a list with 1 item: "OtherBear".
+
+.. note::
+    The list must have classes of the bear itself, not the name as a string.
+
+coala searches for the ``get_dependencies`` function before executing
+the ``DependentBear`` and runs all the Bears which are returned by it.
+
+After running these bears coala gives all the results returned by the Bears
+in the ``dependency_results`` dictionary, which has the Bear's name as a key
+and a list of results as the value, like in this case, we would have
+``dependency_results ==
+{'OtherBear' : [list containing results of OtherBear]]}``.
+
+.. note::
+    ``dependency_results`` is a keyword here and it cannot be called by
+    any other name.
+
+Hidden Results
+--------------
+Apart from regular Results, coala provides HiddenResults, which are used
```

Comment:
@abhsag24  Is `HiddedResults` a keyword? If not maybe add a space in between. :)